### PR TITLE
Travis speed improvement: from 8 to 3 minutes

### PR DESCRIPTION
**Pull request by reinout**
So, the travis run time is down from 8 minutes to 3 minutes!
I got it working with some proper caching, which hopefully is explicit enough.
When there's a change in dockerfile, compose file, requirements.txt: then this is detected and travis re-builds the image and stores it afresh in the cache. (Due to the time-consuming caching, the travis run time takes 11 minutes instead of the previous 8 in this case, but that seems like a good price to pay).

I hope to make some further optimizations in an upcoming pull request so that the docker size goes down, reducing the run time even further.

To get everything working both locally and on travis, I also had to fix the user/group handling. At least, with the README instructions, I couldn't get it to work.
I've changed the uid/gid handling to what's standard now in our cookiecutter: that turned out to work OK.

I've updated the readme with local development instructions, so that ought to be clear enough.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,41 @@
 language: generic
 dist: xenial
 
-env:
-    - UID=1000 USER=qgis_user
```
Comment (reinout) [inline .travis.yml:5]: Note: on travis, the UID is 2000. To my surprise, I must add.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+  # If the docker checksum matches, use the cached docker image.
+  - >
+     md5sum --check $HOME/docker-cache/docker-checksum &&
+     cat $HOME/docker-cache/our-image.tar | docker load || true
+
+install:
+  # Rebuild the docker image, but only if the docker checksum does not match.
+  - >
+     md5sum --check $HOME/docker-cache/docker-checksum ||
+     docker-compose build --build-arg uid=`id -u` --build-arg gid=`id -g`
```
Comment (reinout) [inline .travis.yml:21]: I feed in uid/gid just like in local development. Reason: travis' user id is 2000, not the default 1000.

File: Docker/qgis3.4.5/Dockerfile [hunk 3]
```diff
@@ -7,9 +7,9 @@ ENV DEBCONF_NONINTERACTIVE_SEEN true
 
 ENV TZ Europe/Amsterdam
 
-ARG USER_ID
-ARG GROUP_ID
-ARG USERNAME
+ARG uid=1000
```
Comment (reinout) [inline Docker/qgis3.4.5/Dockerfile:10]: This style is consistent with what's in the cookiecutter.

File: Docker/qgis3.4.5/Dockerfile [hunk 4]
```diff
@@ -74,16 +74,9 @@ RUN pip3 uninstall h5py -y
...
 RUN pip3 install --no-binary=h5py h5py
 
-RUN groupadd -g $GROUP_ID $GROUP_ID
-RUN useradd -m -l -d /home/$USERNAME --shell /bin/bash --uid $USER_ID --gid $GROUP_ID $USERNAME
-
 # Add the qgis-core plugins (like 'db_manager') to the PYTHONPATH
 ENV PYTHONPATH /usr/share/qgis/python/plugins
 
-COPY Docker/docker-entrypoint.sh /
-RUN chmod u+x /docker-entrypoint.sh
```
Comment (reinout) [inline Docker/qgis3.4.5/Dockerfile:84]: The docker-entrypoint stuff has been replaced by the `USER nens` line at the bottom.

**Review by RichardB9 — APPROVED** [review]
Nice! Much simpler and faster than before!

Also learned some new things about docker and travis :)

**Pull request merged**